src/com/sap/piper/PiperGoUtils.groovy

```diff
@@ -52,11 +52,15 @@ class PiperGoUtils implements Serializable {
 
     private boolean downloadGoBinary(url) {
 
-        def httpStatus = steps.sh(returnStdout: true, script: "curl --insecure --silent --location --write-out '%{http_code}' --output ./piper '${url}'")
+        try {
+            def httpStatus = steps.sh(returnStdout: true, script: "curl --insecure --silent --location --write-out '%{http_code}' --output ./piper '${url}'")
 
-        if (httpStatus == '200') {
-            steps.sh(script: 'chmod +x ./piper')
-            return true
+            if (httpStatus == '200') {
+                steps.sh(script: 'chmod +x ./piper')
+                return true
+            }
+        } catch(err) {
+            return false
```

Comment:
Why is swallowing the exception here the better way compared to the old approach of letting the error bubble up? The caller cannot distiguish now if there was a return code `!= 200` or some internal trouble inside the plugin providing the `sh` step or with `curl itself`. What about improving exception handling on the callers side?

In case there is a reason for swallow the exception here the return statement is not needed since we reach in this case automatically the `return false` below.

Reply:
It is to keep the existing behavior and fix the issue of unhandled exceptions with the result of throwing an error without graceful handling.

The existing behavior is that downloadGoBinary returns `true` if the bin was downloaded and `false` if not.
We expect that a download can fail. Resilience is implemented above.

We will have an exception in case the binary cannot be downloaded `Download of Piper go binary failed.`.
In addition the log will show what went wrong.